### ISSUE TITLE
Windows disappear when switching desktops with “App Icons” enabled

### ISSUE THREAD

**Issue by brunohilt** [lwouis/alt-tab-macos#3950]
When you use the “App Icons” appearance and enable the “Show in switcher: Applications” setting, there’s a bug that with displayed windows when switching desktops.

Breakdown:

1. You have an app (e.g., Safari) open on one desktop.
2. On a different desktop, you also have another Safari window open.
3. If you switch between desktops, Alt Tab stops showing the Safari app. This makes it seem like Safari has "disappeared".
4. To make Safari visible again in the app switcher, you must manually focus one of Safari’s windows using something like Mission Control.

In short, the app switcher doesn’t handle multiple app windows across desktops correctly in this setup. It “loses track” of the app until you manually refocus one of its windows.

Video demonstration:

https://github.com/user-attachments/assets/889ae336-884f-4f2c-b98c-20e21cfabc1e

**Comment by lwouis** [comment]
Hi,

Thank you for reporting this issue.

I suspect that the root cause is about how we pick the "main" window of an app, when in `Show: Applications` mode. I've made a fix in [this local build](https://github.com/user-attachments/files/18053282/AltTab.zip). Could you please try it out and let me know if it works for you?

Furthermore, could you please share your preferences? You can run this command in Terminal.app to list them and copy-paste the output here.

`defaults read com.lwouis.alt-tab-macos`

Thank you 🙇

**Comment by brunohilt** [comment]
Hi! Sure, attached. I'll test the build when I get a chance.

I'm also making a separate issue right now about something related, also with these settings - not necessarily a bug report, but might be related to the picking logic. It'll be up in a moment.

```
{
    MSAppCenter310AppCenterUserDefaultsMigratedKey = 1;
    MSAppCenter310CrashesUserDefaultsMigratedKey = 1;
    MSAppCenterAppDidReceiveMemoryWarning = 1;
    MSAppCenterInstallId = "60BED744-CEB9-49DA-B4D4-F0CFC622B12A";
    MSAppCenterNetworkRequestsAllowed = 0;
    MSAppCenterPastDevices = {length = 1149, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 00000389 };
    MSAppCenterSessionIdHistory = {length = 460, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 00000156 };
    MSAppCenterUserIdHistory = {length = 455, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 00000151 };
    NSNavPanelExpandedSizeForOpenMode = "{800, 448}";
    NSOSPLastRootDirectory = {length = 588, bytes = 0x626f6f6b 4c020000 00000410 30000000 ... 10010000 00000000 };
    "NSStatusItem Visible Item-0" = 0;
    "NSWindow Frame NSNavPanelAutosaveName" = "560 406 800 448 0 0 1920 1055 ";
    "NSWindow Frame SUStatusFrame" = "373 367 400 135 0 0 1920 1055 ";
    "NSWindow Frame SUUpdateAlert" = "650 492 620 398 0 0 1920 1055 ";
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2024-12-05 23:53:21 +0000";
    SUUpdateGroupIdentifier = 2207754243;
    SUUpdateRelaunchingMarker = 0;
    alignThumbnails = 1;
    appearanceSize = 0;
    appearanceStyle = 1;
    appearanceTheme = 2;
    appearanceVisibility = 1;
    appsToShow2 = 0;
    blacklist = "[{\\"ignore\\":\\"0\\",\\"bundleIdentifier\\":\\"com.McAfee.McAfeeSafariHost\\",\\"hide\\":\\"1\\"},{\\"hide\\":\\"2\\",\\"bundleIdentifier\\":\\"com.apple.finder\\",\\"ignore\\":\\"0\\"},{\\"hide\\":\\"0\\",\\"bundleIdentifier\\":\\"com.microsoft.rdc.macos\\",\\"ignore\\":\\"2\\"},{\\"bundleIdentifier\\":\\"com.teamviewer.TeamViewer\\",\\"ignore\\":\\"2\\",\\"hide\\":\\"0\\"},{\\"bundleIdentifier\\":\\"org.virtualbox.app.VirtualBoxVM\\",\\"hide\\":\\"0\\",\\"ignore\\":\\"2\\"},{\\"bundleIdentifier\\":\\"com.parallels.\\",\\"hide\\":\\"0\\",\\"ignore\\":\\"2\\"},{\\"bundleIdentifier\\":\\"com.citrix.XenAppViewer\\",\\"hide\\":\\"0\\",\\"ignore\\":\\"2\\"},{\\"bundleIdentifier\\":\\"com.citrix.receiver.icaviewer.mac\\",\\"ignore\\":\\"2\\",\\"hide\\":\\"0\\"},{\\"hide\\":\\"0\\",\\"bundleIdentifier\\":\\"com.nicesoftware.dcvviewer\\",\\"ignore\\":\\"2\\"},{\\"bundleIdentifier\\":\\"com.vmware.fusion\\",\\"ignore\\":\\"2\\",\\"hide\\":\\"0\\"},{\\"bundleIdentifier\\":\\"com.apple.ScreenSharing\\",\\"hide\\":\\"0\\",\\"ignore\\":\\"2\\"},{\\"hide\\":\\"0\\",\\"ignore\\":\\"2\\",\\"bundleIdentifier\\":\\"com.utmapp.UTM\\"},{\\"ignore\\":\\"0\\",\\"hide\\":\\"2\\",\\"bundleIdentifier\\":\\"com.apple.mail\\"}]";
    crashPolicy = 2;
    fadeOutAnimation = true;
    hideAppBadges = true;
    hideColoredCircles = true;
    hideSpaceNumberLabels = false;
    hideStatusIcons = false;
    hideWindowlessApps = true;
    holdShortcut = "\\U2303";
    menubarIcon = 3;
    nextWindowShortcut2 = "\\U21e5";
    preferencesVersion = "7.7.0";
    previewFocusedWindow = false;
    screensToShow = 0;
    showAppsOrWindows = 0;
    showHiddenWindows = 2;
    showHiddenWindows2 = 2;
    showMinimizedWindows = 1;
    showMinimizedWindows2 = 1;
    showOnScreen = 0;
    showTabsAsWindows = false;
    showTitles = 0;
    spacesToShow = 1;
    spacesToShow2 = 0;
    titleTruncation = 2;
    updatePolicy = 1;
    windowDisplayDelay = 100;
    windowOrder = 0;
    windowOrder2 = 0;
}
```

**Comment by brunohilt** [comment]
Tested your local build, it seems to fix this issue!
